### DISCUSSION TITLE
What's the difference between using a PAT vs GCM?

### DISCUSSION THREAD

**Discussion by wAuner** [git-ecosystem/git-credential-manager#423] [Q&A] [answered]
I couldn't understand the difference between the usage of a personal access token which I create manually via the web UI vs using GCM. I'm on macOS.
As far as I can tell, GCM creates a Personal Access Token itself.
Is there any difference from a security perspective between those two approaches?

**Comment by mjcheetham** [comment, marked as answer]
GCM creates OAuth tokens automatically when Git requires a fresh credential, and stores them in the login Keychain (for macOS). It also deals with deleting expired or revoked credentials (Git tells GCM the token is "bad").

Tokens created by GCM have the `repo`, `gist`, and `workflow` scopes.

Creating a PAT in the web UI and using this as the `password` (and using the `osxkeychain` credential helper) would achieve much of the same thing, but you'd need to manually generate the PAT with the correct permissions, and ensure it was up-to-date/generate new ones.

**Reply by wAuner** [reply]
thanks for the info!

**Reply by John0King** [reply]
I wonder how git invoke this `GCM.exe`   how git knows that the GCM is installed (or other tool is installed)

**Reply by ldennington** [reply]
@John0King - GCM is set as a [credential helper](https://git-scm.com/docs/gitcredentials#Documentation/gitcredentials.txt-helper) in users' global Git config files, which is how Git knows to use it.

**Comment by P-Pk** [comment]
ซัพพลอตครับ
